src/main/java/org/embulk/filter/encrypt/EncryptFilterPlugin.java

```diff
@@ -178,35 +247,138 @@ public void transaction(ConfigSource config, Schema inputSchema,
     {
         PluginTask task = config.loadConfig(PluginTask.class);
 
-        if (task.getAlgorithm().useIv() && !task.getIvHex().isPresent()) {
-            throw new ConfigException("Algorithm '" + task.getAlgorithm() + "' requires initialization vector. Please generate one and set it to iv_hex option.");
+        validateAndResolveKey(task, inputSchema);
+
+        control.run(task.dump(), inputSchema);
+    }
+
+    @VisibleForTesting
+    public Map<String, String> retrieveKey(final String bucket, final String path, final AmazonS3 client)
+    {
+        S3Object fullObject = null;
+
+        try {
+            fullObject = client.getObject(new GetObjectRequest(bucket, path));
```

Comment:
If getObject return a null (if there're no object under `path`), I think we should fail-fast and throw a meaningful message, otherwise it will be a NPE when we try to `getObjectContent()`

Reply:
Yeah, it would be better. 👍